starter-code/src/QueueDataStructure.js

```diff
@@ -1,2 +1,39 @@
-function QueueDataStructure () {
-}
+class QueueDataStructure {
+
+    constructor() {
+        this.queueControl = [];
+        this.MAX_SIZE = 10;
+    }
+
+    isEmpty() {
+        if (this.queueControl.length === 0) {
+            return true
+        } else {
+            return false;
+        }
+    }
+
+    canEnqueue() {
+        if (this.queueControl.length < this.MAX_SIZE) {
+            return true
+        }
+        return false
+    }
+
+    enqueue(element) {
+        if (this.canEnqueue()) {
+            this.queueControl.unshift(element)
+            return this.queueControl;
+        } else {
+            return 'Queue Overflow';
+        }
+    }
+
+    dequeue() {
+        if (!this.isEmpty()) {
+            return this.queueControl.pop()
+        } else {
+            return 'Queue Underflow';
+        }
```

Comment:
You can also use this ternary expression "return !this.isEmpty() ? this.queueControl.pop() : 'Queue Underflow';"